### PR TITLE
Fixing GraphQL SSE crash with large stanza payloads

### PR DESCRIPTION
**Pull request by JanuszJakubiec**
**Fix**: Wrap jiffy:encode in iolist_to_binary to handle long messages.

**Issue**: When encoding a message to JSON that exceeds 2048 characters, jiffy:encode returns a list of binaries instead of a single binary.

**Solution**: By wrapping jiffy:encode in iolist_to_binary, we ensure that the encoded output is always a single binary, resolving the issue with long messages.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236164) / elasticsearch_and_cassandra_mnesia / 32151fb32c3e8c528c2b297fb21055948200266a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4364/236164/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4364/236164/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%40b85717d49703.2024-08-23_10.43.06/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

File: src/graphql/mongoose_graphql_response.erl [hunk 1]
```diff
@@ -3,10 +3,10 @@
 -export([term_to_json/1, term_to_pretty_json/1]).
 
 term_to_json(Term) ->
-    jiffy:encode(fixup(Term)).
+    iolist_to_binary(jiffy:encode(fixup(Term))).
```
Comment (arcusfelis) [inline src/graphql/mongoose_graphql_response.erl:6]: o, nice.
I've checked and `iolist_to_binary(Binary)` would not copy Binary, so looks good.

File: big_tests/tests/graphql_sse_SUITE.erl [hunk 2]
```diff
@@ -145,6 +145,10 @@ sse_should_not_get_timeout(Config) ->
         timer:sleep(2000),
         escalus:send(Bob, escalus_stanza:chat(From, To, <<"Hello again!">>)),
         sse_helper:wait_for_event(Stream),
+        timer:sleep(2000),
```
Comment (arcusfelis) [inline big_tests/tests/graphql_sse_SUITE.erl:148]: Do we need to have these sleeps?
We could also made a separate testcase like `sse_works_with_long_messages`.

And run sse_should_not_get_timeout and sse_works_with_long_messages in parallel.

**Review by arcusfelis — COMMENTED** [review]
Reviewed, you could split the testcase into two.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236265) / elasticsearch_and_cassandra_mnesia / 398fffd28fa25872c562b2d70ed3e6b8ad6eb432
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4364/236265/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4364/236265/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%403a77b6bbd275.2024-08-26_13.49.27/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Review by arcusfelis — APPROVED** [review]
ok

**Pull request merged**